### PR TITLE
box initial ruleset

### LINKED ISSUE #95: Initial rule set: Box
Panther supports Box logs as a SaaS data source. We should add some initial rules to run against this data source. Let's aim for 5 rules to get off the ground.

Reference: https://docs.runpanther.io/log-analysis/supported-logs/box

Since box is a file sharing service, lets focus on rules about overly permissive sharing activities similar to the G Suite rules.

### PR DESCRIPTION
**Pull request by lindsey-w**
### Background

Initial set of rules for `Box.Events` data source. Closes #95 

### Changes

* New rules and tests cases for box events

### Testing

* `make ci`

### CODE REVIEW

**Review by nhakmiller — COMMENTED** [review]
These seem like some good baseline rules. I would like to suggest these rules as well:

Return True on `ITEM_DOWNLOAD`, dedup by user, threshold something big like 50 or 100. Threshold period moderate like say 1 hour. Looking for mass data exfil type activity.

Similar as above, but for any permissions changes to documents.

And finally a more ops related rule, but for whenever a file gets created/uploaded, check the `space_amount` and  `space_used` fields in the `source` and alert if `space_used` is over 90% of `space_amount`. Dedup on user, max dedup period. Probably Low severity.

My hope with these is just to have 5 rules that don't depend on box shield, which as I understand is a rather expensive and optional piece of box logging.

**Comment by nhakmiller** [comment]
> And finally a more ops related rule, but for whenever a file gets created/uploaded, check the `space_amount` and `space_used` fields in the `source` and alert if `space_used` is over 90% of `space_amount`. Dedup on user, max dedup period. Probably Low severity. (I don't know if these will always be present, just looking at the events docs on box's website)

Actually ignore this one, I'm not seeing this show up in other places in the docs and it doesn't show up in our docs. The other two are fine.

File: box_rules/box_access_granted.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+RuleID: Box.Access.Granted
+DisplayName: Box Access Granted
+Enabled: true
+LogTypes:
+  - Box.Event
+Tags:
+  - Box
+Severity: Low
+Description: >
+  A user granted access to their box account.
```
[the remark this answers is not in the document]
Comment (lindsey-w) [reply box_rules/box_access_granted.yml:12]: I misunderstood this event type, and it turns out it is logged when Box support is given access to a user's account.  I can keep it in and add some additional context here - or I can remove it completely? IMO, I might still be interested in this event (depending on what actions are taken when someone was using it).

File: box_rules/box_access_granted.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+Tags:
+  - Box
+Severity: Low
+Description: >
+  A user granted access to their box account. 
+Reference: https://developer.box.com/reference/resources/event/
+Runbook: >
+  Investigate whether the user purposefully granted access to their account. 
+SummaryAttributes:
+  - event_type
```
[the remark this answers is not in the document]
Comment (lindsey-w) [reply box_rules/box_access_granted.yml:17]: Good point! I'll add ip_address. Most of the fields are nested so I don't think they can be accessed by SummaryAttributes quite yet

File: box_rules/box_new_login.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+def rule(event):
+    return (event.get('event_type') == 'ADD_LOGIN_ACTIVITY_DEVICE' or
+            event.get('event_type') == 'DEVICE_TRUST_CHECK_FAILED	')
```
[the remark this answers is not in the document]
Comment (lindsey-w) [reply box_rules/box_new_login.py:3]: I added the difference in comments, but ADD_LOGIN_ACTIVITY_DEVICE == user logging in with a new device fingerprint and DEVICE_TRUST_CHECK_FAILED == user tried to login from a device that doesn't meet the requirements (OS version, in the Domain, anitvirus installed, etc.).  After writing this out... I decided they are two distinct things and split them into two different rules.

**Review by nhakmiller — APPROVED** [review]
LGTM

**Pull request merged**